test/ctr.bats

```diff
@@ -12,15 +12,17 @@ function teardown() {
 	cleanup_test
 }
 
+function is_v2() {
+	test "$(stat -f -c%T /sys/fs/cgroup)" = "cgroup2fs"
+}
+
 function wait_until_exit() {
     ctr_id=$1
     # Wait for container to exit
     attempt=0
     while [ $attempt -le 100 ]; do
         attempt=$((attempt + 1))
-        run crictl inspect -o table "$ctr_id"
-        echo "$output"
-        [ "$status" -eq 0 ]
+        output=$(crictl inspect -o table "$ctr_id")
```

Comment:
In at least the Buildah and Podman repos, we've been moving these calls into a helper function.  Changing calls directly to the tool to `run_buildah` as it gives us some extra error handling.   Might be something to consider for cri-o. @edsantiago has driven that in those projects, I'd be interested in hearing his opinion on these changes.  Also, I'm not seeing a status check here, should there be?

Reply:
bats runs test with `set -e`, so the status check is implicit and one can run simple tests like
```bash
@test "my test" {
   command1 args ...
   command2 args ...
   [ -f output.file ]
}
```

rather than
```bash
@test "my test" {
   run command1 args ...
   echo "$output"
   [ "$status" -eq 0 ]

   run command2 args ...
   echo "$output"
   [ "$status" -eq 0 ]

   if [ ! -f output.file ]; then
      exit 1
   fi
```

These two examples are roughly equivalent, with some minor details, and I strongly prefer the former -- for clarity.

Besides, there are many inherent problems with `run`, some of them I tried to document in https://github.com/bats-core/bats-core/pull/343 and https://github.com/bats-core/bats-core/pull/366.

I guess I'll eventually come with a doc (for test directory) explaining how to use and not to use `run`.

Reply:
> In at least the Buildah and Podman repos, we've been moving these calls into a helper function. Changing calls directly to the tool to run_buildah as it gives us some extra error handling. Might be something to consider for cri-o.

@TomSweeneyRedHat can you please share a link with an example of such changes?

Reply:
https://github.com/containers/buildah/blob/50be5719e09243b4d30a248a72eb6fd2b217b45a/tests/helpers.bash#L105-L107

This has repeatedly proven itself valuable over the years. The key aspects are that it does exit-status check automatically, logs the command run and its output (only displayed when there's an error), and there's the `expect_output` helper (further below) for string checking. In regular BATS all you get is "this did not match", but with no indication of what the actual values were; which makes it completely useless for diagnosing.

I've long had a PR planned for submission to BATS itself, altering `run` itself, but have never finished wrapping it up. Maybe if there's ever a quiet day...

Reply:
What @edsantiago said.  Thanks Ed!

Reply:
I've submitted my `run` modifications to the project: https://github.com/bats-core/bats-core/pull/367

Reply:
Thanks @edsantiago @TomSweeneyRedHat, let's continue this discussion in  bats-core/bats-core#367